pkg/controller/instance/instance_controller.go

```diff
@@ -376,24 +423,46 @@ type ReconcileInstance struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=kudo.dev,resources=instances,verbs=get;list;watch;create;update;patch;delete
 func (r *ReconcileInstance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
 	// Fetch the Instance instance
-	instance := &kudov1alpha1.Instance{}
-	err := r.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
+	// Fetch the Instance instance
+	instance, err := getInstance(ctx, r, request)
+	if err != nil || instance == nil {
+		return reconcile.Result{}, err
```

Comment:
so here we retry when instance which we are reconciling does not exist? 🤔 is that correct? or will that again lead to infinite loop of reconciles.

Reply:
first that was the original logic I believe... it is just more clear now
second... I do believe that is the k8s controller way.   It is worth understanding deeper for sure.    
As this was represented in the original code: https://github.com/kudobuilder/kudo/blob/master/pkg/controller/instance/instance_controller.go#L386
I hope this isn't a block on this PR... it is definitely easier to read and removes logic out of the predicate... that is the focus of this PR

Reply:
it is also a great opportunity to reason about the code!   so thanks!